tools/wptserve/wptserve/request.py

```diff
@@ -610,8 +607,33 @@ def from_field_storage(cls, fs):
         return self
 
 
+class BinaryCookieParser(BaseCookie):
+    """A subclass of BaseCookie that returns values in binary strings
+
+    This is not intended to store the cookies; use Cookies instead."""
+
+    def value_decode(self, val):
+        """Decode value from network to (real_value, coded_value).
+
+        Override BaseCookie.value_decode."""
+        real = val.encode('iso-8859-1') if isinstance(val, text_type) else val
+        return real, val
+
+    def value_encode(self, val):
+        raise NotImplementedError('BinaryCookieParser is not for setting cookies')
```

Comment:
Why not? How would you set a value to 0x80?

Reply:
This is only used in `Request`, and only as a parser. First, `request.cookies` are not intended to be changed (you'd want to change `response.cookies` instead); second, even if you do, `request.cookies` is the `Cookies` class not this class (this parser class gets discarded immediately).